src/clang_parser.cpp

```diff
@@ -672,11 +661,40 @@ bool ClangParser::parse(ast::Program *program, BPFtrace &bpftrace, std::vector<s
     return false;
   }
 
-  if (!handler.check_diagnostics(input))
+  std::vector<std::string> error_msgs;
+  if (!handler.check_diagnostics(input, error_msgs, true))
+  {
+    for (auto &msg : error_msgs)
+    {
+      if (get_unknown_type(msg) != "" && !bpftrace.force_btf_)
+      {
+        LOG(ERROR) << "Try running with --btf to force BTF processing or "
+                      "include headers with missing type definitions";
+      }
+    }
     return false;
+  }
 
   CXCursor cursor = handler.get_translation_unit_cursor();
   return visit_children(cursor, bpftrace);
 }
 
+/*
+ * Parse the given Clang diagnostics message and if it has the form:
+ *   unknown type name 'type_t'
+ * return type_t.
+ */
+std::string ClangParser::ClangParser::get_unknown_type(
```

Comment:
Returning an optional instead of empty string would be cleaner API
```suggestion
std::optional<std::string> ClangParser::ClangParser::get_unknown_type(
```